code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -296,3 +296,135 @@
+	//rand doesn't seem to work with n^-e
+	OptimalTempMin 		+= rand(-100, 100)
+	OptimalTempMax 		= (OptimalTempMin+rand(0, 200))
+	ExplodeTemp 		= (OptimalTempMax+rand(0, 200))
+	OptimalpHMin 		+= rand(1, 10)
+	OptimalpHMax 		= (OptimalpHMin + rand(1, 5))
+	ReactpHLim 			+= rand(-2, 2)
+	CurveSharpT 		+= (rand(1, 500)/100)
+	CurveSharppH 		+= (rand(1, 500)/100)
+	ThermicConstant		+= rand(-50, 50)
+	HIonRelease 		+= (rand(-25, 25)/100)
+	RateUpLim 			+= (rand(1, 1000)/100)
+	PurityMin 			+= (rand(-1, 1)/10)
+	var/additions = list("aluminum", "silver", "gold", "plasma", "silicon", "bluespace", "uranium", "milk")
+	required_reagents[pick(additions)] = rand(1, 5)//weird
+
+/datum/chemical_reaction/fermi/secretcatchem/FermiFinish(datum/reagents/holder, var/atom/my_atom)//Strange how this doesn't work but the other does.
+	message_admins("Someone found the hidden reaction. Amazing!! Please tell Fermis!!")
+
+//ReagentVars
+//Turns you into a cute catto while it's in your system.
+//If you manage to gamble perfectly, makes you a catgirl after you transform back. But really, you shouldn't end up with that with how random it is.
+/datum/reagent/fermi/secretcatchem //Should I hide this from code divers? A secret cit chem?
+	name = "secretcatchem" //an attempt at hiding it
+	id = "secretcatchem"
+	description = "An illegal and hidden chem that turns people into cats. It's said that it's so rare and unstable that having it means you've been blessed."
+	taste_description = "hairballs and cream"
+	color = "#ffc224"
+	var/catshift = FALSE
+	var/mob/living/simple_animal/pet/cat/custom_cat/catto = null
+
+/datum/reagent/fermi/secretcatchem/New()
+	name = "Catgirli[pick("a","u","e","y")]m [pick("apex", "prime", "meow")]"
+
+/datum/reagent/fermi/secretcatchem/on_mob_add(mob/living/carbon/human/H)
+	. = ..()
+	var/current_species = H.dna.species.type
+	var/datum/species/mutation = /datum/species/human/felinid
+	if((mutation != current_species) && (purity >= 0.8))//ONLY if purity is high, and given the stuff is random. It's basically impossible to get this to 1. It already requires felind too, so no new functionality there.
+		H.set_species(mutation)
+		H.gender = FEMALE
+		//exception(al) handler:
+		H.dna.features["mam_tail"] = "Cat"
+		H.dna.features["tail_human"] = "Cat"
+		H.dna.features["ears"]  = "Cat"
+		H.dna.features["mam_ears"] = "Cat"
+		H.dna.features["tail_lizard"] = "Cat"
+		H.dna.features["mam_tail"] = "Cat"
+		H.dna.features["mam_tail_animated"] = "Cat"
+		H.facial_hair_style = "Shaved"
+		H.verb_say = "mewls"
+		catshift = TRUE
+		playsound(get_turf(H), 'modular_citadel/sound/voice/merowr.ogg', 50, 1, -1)
+	to_chat(H, "<span class='notice'>You suddenly turn into a cat!</span>")
+	catto = new(get_turf(H.loc))
+	H.mind.transfer_to(catto)
+	catto.name = H.name
+	catto.desc = "A cute catto! They remind you of [H] somehow."
+	catto.color = "#[H.dna.features["mcolor"]]"
+	H.forceMove(locate(10,6,1))//To the zelda room.

```

Comment:
Please do not hardcode map coordinates into procs. And please, absolutely do not intentionally place mobs within the nullspace room; a far better way to go about hiding the mob here would be to do an `H.forceMove(catto)`
```suggestion
	H.forceMove(catto)
```

Reply:
I need to hide the H, I tried nullspace, but it put them in the zelda room everytime.

Reply:
Did you try `H.moveToNullspace()` yet?

Reply:
I think that's the proc I tried. I'll try using that and testing it, tomorrow when I start going over the code reviews though!

Reply:
well, give a look at how the polymorph spell works then.